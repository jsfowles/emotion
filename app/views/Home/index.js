import React from 'react';
import glamorous from 'glamorous';

import MarkdownRenderer from '@components/MarkdownRenderer';
import { Main } from '@identity/wrappers';
import README from '../../../README.md';

export default () => (
  <Main spacing={{ padding: 'vertical-lg' }}>
    <MarkdownRenderer source={ README } />
  </Main>
);
