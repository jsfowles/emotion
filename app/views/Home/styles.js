import styled, { keyframes } from 'react-emotion';

export const Bounce = keyframes`
0% {
  background-position: 0% 50%
}
50% {
  background-position: 100% 50%
}
100% {
  background-position: 0% 50%
}
`;

export const Border = styled('div')({
  height: '500px',
  width: '900px',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  flexDirection: 'column',
  background: 'black',
  position: 'relative',
  zIndex: 2,
  borderRadius: '5px',
});

export const Div = styled('div')(
  {
    background: '#1D1F20',
    borderWidth: '3px',
    position: 'relative',
    borderRadius: '5px',
    '&:after': {
      content: '""',
      position: 'absolute',
      top: 'calc(-1 * 3px)',
      left: 'calc(-1 * 3px)',
      height: 'calc(100% + 3px * 2)',
      width: 'calc(100% + 3px * 2)',
      background:
        'linear-gradient(60deg, #f79533, #f37055, #ef4e7b, #a166ab, #5073b8, #1098ad, #07b39b, #6fba82)',
      borderRadius: 'calc(2 * 3px)',
      zIndex: 1,
      animation: `${Bounce} 3s ease alternate infinite`,
      backgroundSize: '300% 300%',
    },
  },
  props => ({ backgroundColor: props.color }),
);

export const Emotion = styled('h1')(
  {
    color: 'hotpink',
    fontFamily: 'Streamster',
    lineHeight: '150px',
    letterSpacing: '4px',
  },
  props => ({ fontSize: props.fontSize }),
);

export const Main = styled('div')({
  height: '100vh',
  width: '100vw',
  backgroundColor: 'black',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  flexDirection: 'column',
});
