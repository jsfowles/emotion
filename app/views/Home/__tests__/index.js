import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from '@views/Home';

jest.mock('../../../../README.md', () => 'Hello World');

describe('Home View', () => {
  it('renders without any props', () => {
    const wrapper = shallow(<Home />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
