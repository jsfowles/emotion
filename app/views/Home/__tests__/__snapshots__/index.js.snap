// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View renders without any props 1`] = `
<glamorous(main)
  spacing={
    Object {
      "padding": "vertical-lg",
    }
  }
>
  <Component
    source="Hello World"
  />
</glamorous(main)>
`;
